# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/mac/MOXTextMarkerDelegate.h
# Commit: 0441d1553b6b
# Full Hash: 0441d1553b6bbf99d711300680eccd335e0b3545
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-07-03 03:56:55
# Regressor Bug: 1649575
# File Overlap Count: 1
# Description:
#   Bug 1649575 - Part 3: Introduce MOXTextMarkerDelegate. r=morgan
#   
#   The delegate is associated with accessible documents (either local or remote).
#   
#   I made a separate protocol for all the text marker stuff as it really is a seperate API.
# ==============================================================================

diff -r 65f8e705d730 -r 0441d1553b6b accessible/mac/MOXTextMarkerDelegate.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/accessible/mac/MOXTextMarkerDelegate.h	Thu Jul 02 21:33:07 2020 +0000
@@ -0,0 +1,37 @@
+/* -*- Mode: Objective-C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import <Cocoa/Cocoa.h>
+
+#import "MOXAccessibleProtocol.h"
+
+#include "AccessibleOrProxy.h"
+
+@interface MOXTextMarkerDelegate : NSObject <MOXTextMarkerSupport> {
+  mozilla::a11y::AccessibleOrProxy mGeckoDocAccessible;
+}
+
++ (id)getOrCreateForDoc:(mozilla::a11y::AccessibleOrProxy)aDoc;
+
++ (void)destroyForDoc:(mozilla::a11y::AccessibleOrProxy)aDoc;
+
+- (id)initWithDoc:(mozilla::a11y::AccessibleOrProxy)aDoc;
+
+// override
+- (id)moxStartTextMarker;
+
+// override
+- (id)moxEndTextMarker;
+
+// override
+- (NSNumber*)moxLengthForTextMarkerRange:(id)textMarkerRange;
+
+// override
+- (NSString*)moxStringForTextMarkerRange:(id)textMarkerRange;
+
+// override
+- (id)moxTextMarkerRangeForUnorderedTextMarkers:(NSArray*)textMarkers;
+
+@end